### ISSUE TITLE
Local asset mapping: FeedMe looks in the wrong folder

### ISSUE THREAD

**Issue by hiasl** [craftcms/feed-me#833]
### Description
- I am importing entries from a Google Spreadsheet. 
- The spreadsheet has one column "image", which contains a filename only (my-image.jpg). The file being referenced already exists in the Craft system, where it is located in the root of a public volume "images". 
- The system has a second volume "documents". The asset field I am trying to assign the image to has set both (all) volumes as possible source, but the _default upload location_ is set to the "documents" share.
- When importing, FeedMe shows the following in the log: 
_Search for existing asset with query_ `{"siteId":1,"status":null,"folderId":1,"kind":null,"limit":"9","filename":"my-image.jpg","includeSubfolders":true}`
- The problem: the image is available, but not in `folderId: 1`. Instead, `folderId: 2` would be correct (which is the base folder of the "images" share). The image is not found and will not be correctly assigned.
- When I change the default upload location for the asset field to be in the "images" share, it works.

My conclusion: It is wrong to resolve the `folderId` from the _default upload location_ of the asset field.. It should be taken from the selected volumes for the asset field and it should also contain more than one `folderId` if more than one volume is configured.

### Additional info

- Craft version: 3.5.19.1
- PHP version: 7.4
- Plugins & versions: FeedMe 4.3.6

**Comment by therealpecus** [comment]
this bug or quirk is almost a year old but I found it because I have the same issue. To avoid a breaking change, there should be a preference that ignores the default upload location for the field being imported so that asset search can span the whole volume.

is this change possible?

setting a custom configuration would mean the user configuring FeedMe is aware of the potential risk with images with the same filename being available both in the root volume — where presumably would be found first — and in a subfolder, that could map the default upload location for the field.

Thanks!

**Comment by RetroChaos** [comment]
This is very frustrating as this bug/oversight is still a problem even in 4.5.3. Is there any way to specify what asset volume / folder to look in on linking assets? Or provide a full path to an image as an option? We use Feed-Me quite a lot but it means each time I have to dump a load of assets in our default folder, link them, then move them to where they should be on the filesystem.

**Comment by sg-modlab** [comment]
Chiming in that we are hitting this issue as well right at the tail end of notable site redesign where we have to migrate a host of content.

**Comment by sander-lameco** [comment]
> My opinion: Search in the default location first (and take the first match from there if there is any), but look in other possible volumes as well if they are configured. This would not be a breaking change and still solve the issue.

I agree with @hiasl, this would be the best way to tackle this. For now we are creating a new field mapping to the volume we want it to search in.
